src/storage/DataAccessorBasedStore.ts

```diff
@@ -347,8 +348,22 @@ export class DataAccessorBasedStore implements ResourceStore {
    * @param slug - Slug to use for the new URI.
    */
   protected createURI(container: ResourceIdentifier, isContainer: boolean, slug?: string): ResourceIdentifier {
-    return { path:
-        `${ensureTrailingSlash(container.path)}${slug ? trimTrailingSlashes(slug) : uuid()}${isContainer ? '/' : ''}` };
+    let path = ensureTrailingSlash(container.path);
+    path += slug ? this.cleanSlug(slug) : uuid();
+    path += isContainer ? '/' : '';
+    return { path };
+  }
+
+  /**
+   * Clean http Slug to be compatible with the server. Makes sure there are no unwanted characters
+   * e.g.: cleanslug('&/%26') returns '%26%2F%26'
+   * @param slug - the slug to clean
+   */
+  protected cleanSlug(slug: string): string {
+    let res = trimTrailingSlashes(slug);
+    res = toCanonicalUriPath(res);
+    res = res.split('/').join(encodeURIComponent('/'));
```

Comment:
Since you are updating the tests to also support the slashes being changed in the slug, it is only a small change to now fully support what is being asked in https://github.com/solid/community-server/issues/574#issuecomment-780540188. In case you missed it in the text, the consensus is to reject slugs that contain a slash. So this function would be something like
```typescript
if (slug.contains('/')) {
  throw new BadRequestHttpError('Slugs should not have slashes');
}
return toCanonicalUriPath(slug);
```
Once the test helper functions have been updated to use PUT as I mentioned there this should not cause more issues in the integration tests. There might be a few unit tests that need to be updated in case they depended on this (wrong) behaviour.

Reply:
While updating the tests i noticed that a PUT(, PATCH and DELETE) request returns 205.
I am used to seeing 201 or 204.
Why did you choose to return status 205 for these requests?

Reply:
201 means: "A resource was created, check the Location header for the location of this new resource"
205 means: "The data at this location has changed".

Which makes sense since with a POST you target the container the new resource will be located in, with a PUT you target the actual resource where the data will be put.

Reply:
@joachimvh I get that conceptually, the URI to which one `PUT`s a new resource has it's data changed (from not existing to existing), but hasn't `201` been the correct respons to successful `PUT` requests for ages? The [HTTP spec](https://tools.ietf.org/html/rfc7231#section-4.3.4) at least does not explicitly suggest `205` as a valid response.

Reply:
Fair point, tagging @RubenVerborgh here as the status codes are based on the original design document at https://github.com/solid/solid-architecture/blob/master/server/request-flow.md

Reply:
> but hasn't `201` been the correct respons to successful `PUT` requests for ages? The [HTTP spec](https://tools.ietf.org/html/rfc7231#section-4.3.4) at least does not explicitly suggest `205` as a valid response.

Neither of those are an obstacle for `205`; but happy with `201` here.

Reply:
Just for clarity my point, I put some emphasis on the definition of the `205` code: 

> The 205 (Reset Content) status code indicates that the server has fulfilled the request and desires that the user agent **reset the 'document view'**, which caused the request to be sent, to its original state as received from the origin server.  This response is intended to support a common data entry **use case where the user receives content that supports data entry** (a form, notepad, canvas, etc.), _enters_ or manipulates data in that space, causes the entered data to be _submitted_ in a request, and then the data entry mechanism is _reset_ for the next entry so that the user can easily initiate another input action. Since the 205 status code implies that no additional content will be provided, **a server MUST NOT generate a payload in a 205 response.**

To me, this clearly indicates why `205` is inappropriate here. In the design document (thanks for the reference, @joachimvh), the text reads: "_The server responds with ... 205 if method is PUT or PATCH. When possible, the body of this response should respect the agent's representation preferences._" This is part of "_Step 5: Return a representation_," and thus clearly not in line with the no-payload directive of the `205` code. Moreover, since the server has no native document view for data entry, there is nothing to be 'reset'. 

The more appropriate choice would therefore, i.m.o., be a `201` response for `PUT` requests, and a `200` for `PATCH`es. This might of course better be another issue.

Reply:
Going to create a separate issue for this as this is mostly irrelevant for this PR.